qtconsole/console_widget.py

```diff
@@ -837,6 +837,24 @@ def _increase_font_size(self):
     def _decrease_font_size(self):
         self.change_font_size(-1)
 
+    def select_all_smart(self):
+        """ Select current cell, or, if already selected, the whole document.
+        """
+        c = self._get_cursor()
+        sel_range = c.selectionStart(), c.selectionEnd()
+
+        c.clearSelection()
+        c.setPosition(self._get_prompt_cursor().position())
+        c.setPosition(self._get_end_pos(),
+                                 mode=QtGui.QTextCursor.KeepAnchor)
+        new_sel_Range = c.selectionStart(), c.selectionEnd()
+        if sel_range == new_sel_Range:
+            # cell already selected, expand selection to whole document
+            self.select_all()
```

Comment:
Let's rename `select_all` to `select_document` to avoid confusion.

Reply:
Fixed